nixos/modules/services/web-apps/stash.nix

```diff
@@ -0,0 +1,504 @@
+        '';
+        type = bool;
+        default = true;
+      };
+      mutablePlugins = mkEnableOption "Whether plugins/themes can be installed, updated, uninstalled manually.";
+      mutableScrapers = mkEnableOption "Whether scrapers can be installed, updated, uninstalled manually.";
+      plugins = pluginType "plugins";
+      scrapers = pluginType "scrapers";
+
+      settings = mkOption {
+        type = settingsType;
+        description = ''Stash configuration'';
+        defaultText = literalExpression ''
+          {
+            blobs_path = "${dataDir}/blobs";
+            cache = "${dataDir}/cache";
+            database = "${dataDir}/stash-go.sqlite";
+            generated = "${dataDir}/generated";
+            plugins_path = if (services.stash.mutablePlugins) then "${dataDir}/plugins" else services.stash.plugins;
+            scrapers_path = if (services.stash.mutableScrapers) then "${dataDir}/scrapers" else services.stash.scrapers;
+
+            blobs_storage = "FILESYSTEM";
+            calculate_md5 = false;
+            create_image_clip_from_videos = false;
+            dangerous_allow_public_without_auth = false;
+            gallery_cover_regex = "(poster|cover|folder|board)\.[^\.]+$";
+            no_proxy = "localhost,127.0.0.1,192.168.0.0/16,10.0.0.0/8,172.16.0.0/12";
+            nobrowser = true;
+            notifications_enabled = true;
+            parallel_tasks = 1;
+            preview_audio = true;
+            preview_exclude_end = 0;
+            preview_exclude_start = 0;
+            preview_segment_duration = 0.75;
+            preview_segments = 12;
+            security_tripwire_accessed_from_public_internet = "";
+            sequential_scanning = false;
+            show_one_time_moved_notification = true;
+            sound_on_preview = false;
+            theme_color = "#202b33";
+            video_file_naming_algorithm = "OSHASH";
+            write_image_thumbnails = true;
+          }
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion =
+          (cfg.settings.username or null != null -> cfg.settings.password or null != null)
+          && (cfg.settings.password or null != null -> cfg.settings.username or null != null);
+        message = "You must set either both username and password, or neither.";
+      }
+    ];
+
+    services.stash.settings = settingsDefault;
+
+    networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ cfg.settings.port ];
+
+    systemd.services.stash = {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      path = [
+        ffmpeg-full
+        python3
+        ruby
+      ];
+      environment.STASH_CONFIG_FILE = "${cfg.dataDir}/config.yml";
+      serviceConfig = {
+        DynamicUser = true;
```

Comment:
Using DynamicUser with services that have a lot of data is encouraged as it basically chowns all the files on every start which can be quite slow

Reply:
That's good to know for future reference.